content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,423 @@
+the MNIST dataset.
+
+You can follow along with the steps here or you can clone the repo directly from
+GitHub and play with it. To clone the repo, run the following commands.
+
+```bash
+git clone https://github.com/iterative/checkpoints-tutorial
+cd checkpoints-tutorial
+```
+
+It is highly recommended you create a virtual environment for this example. You
+can do that by running:
+
+```bash
+python3 -m venv .venv
+```
+
+Once your virtual environment is installed, you can start it up with one of the
+following commands.
+
+- On Mac/Linux: `source .venv/bin/activate`
+- On Windows: `.\.venv\Scripts\activate`
+
+Once you have your environment set up, you can install the dependencies by
+running:
+
+```bash
+pip install -r requirements.txt
+```
+
+This will download all of the packages you need to run the example. Now you have
+everything you need to get started with experiments and checkpoints.
+
+## Setting up a DVC pipeline
+
+If you don't have a DVC pipeline setup in your project, adding one only takes a
+few commands. At the root of the project, run:
+
+```bash
+dvc init
+```
+
+This sets up the files you need for your DVC pipeline to work.
+
+This will create a _dvc.yaml_ file in your root directory.
+
+Now we need to add a stage for training our model within a DVC pipeliene. We'll
+do that with the following command:
+
+```bash
+dvc stage add -n train -d data/MNIST -d train.py -c model.pt --plots-no-cache predictions.json -p seed,lr,weight_decay --live dvclive python train.py
+```
+
+Before we go any further, this is a great point to add these changes to your Git
+history. You can do that with the following commands:
+
+```bash
+git add .
+git commit -m "created DVC pipeline"
+```
+
+## Enabling DVC pipelines for checkpoint experiments
+
+After running the command above to setup your _train_ stage, your _dvc.yaml_
+should have the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory. This will create a new stage with checkpoints enabled, but
+does not update existing stages.
+
+The rest of the command `dvc stage add` sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots and showing the training
+metrics.
+
+We'll manually add a slight modification to the way our plots are handled. Add
```

Comment:
We don't use the confusion matrix in this tutorial, do we? It is better to remove this part and especially yaml modifications (it is not easy to do for users).

Actually the entire section "Enabling DVC pipelines for checkpoint experiments" can be removed. If some explanation is really needed - it can be moved to the previous pipeline creation section.

Reply:
I'd suggest removing the confusion matrix plot from the repository and code. It should be a subject of a different tutorial.